cli/sub_command.go

```diff
@@ -139,89 +145,56 @@ func (c *subCmd) subscribe(_ *fisk.ParseContext) error {
 		}
 
 		ctr++
-		if dump {
-			// Output format 1/3: dumping, to stdout or files
-
-			stdout := c.dump == "-"
-			outFile := ""
-			if !stdout {
-				if info == nil {
-					outFile = filepath.Join(c.dump, fmt.Sprintf("%d.json", ctr))
-				} else {
-					outFile = filepath.Join(c.dump, fmt.Sprintf("%d.json", info.StreamSequence()))
-				}
-			}
-
-			// dont want sub etc
-			msg := nats.NewMsg(m.Subject)
-			msg.Header = m.Header
-			msg.Data = m.Data
-			msg.Reply = m.Reply
-
-			jm, err := json.Marshal(msg)
-			if err != nil {
-				log.Printf("Could not JSON encode message: %s", err)
-			} else if stdout {
-				os.Stdout.WriteString(fmt.Sprintf("%s\000", jm))
-			} else {
-				err = os.WriteFile(outFile, jm, 0600)
-				if err != nil {
-					log.Printf("Could not save message: %s", err)
-				}
-
-				if ctr%100 == 0 {
-					fmt.Print(".")
-				}
-			}
-
-		} else if c.raw {
-			// Output format 2/3: raw
-
-			fmt.Println(string(m.Data))
-
+		if c.match && m.Reply != "" {
+			matchMap[m.Reply] = m
 		} else {
-			// Output format 3/3: pretty
+			printMsg(c, m, nil, ctr)
+		}
 
-			if info == nil {
-				if m.Reply != "" {
-					fmt.Printf("[#%d] Received on %q with reply %q\n", ctr, m.Subject, m.Reply)
-				} else {
-					fmt.Printf("[#%d] Received on %q\n", ctr, m.Subject)
-				}
-			} else if jetStream {
-				fmt.Printf("[#%d] Received JetStream message: stream: %s seq %d / subject: %s / time: %v\n", ctr, info.Stream(), info.StreamSequence(), m.Subject, info.TimeStamp().Format(time.RFC3339))
-			} else {
-				fmt.Printf("[#%d] Received JetStream message: consumer: %s > %s / subject: %s / delivered: %d / consumer seq: %d / stream seq: %d\n", ctr, info.Stream(), info.Consumer(), m.Subject, info.Delivered(), info.ConsumerSequence(), info.StreamSequence())
+		if ctr == c.limit {
```

Comment:
I wonder if we should count a matched reply in this counter, it seems weird that when I do `nats sub --match-replies --count 2 '>'` that I get 1 message with its reply.  Seems intuitive that I should get 2 messages and their replies.

What do you think?